autowheelbarrow.lua

```diff
@@ -0,0 +1,120 @@
+local repeatutil = require("repeat-util")
+local GLOBAL_KEY = "autowheelbarrow"
+enabled = enabled or false
+
+-- Main function for managing wheelbarrows
+local function unassign_wheelbarrows()
+    local wheelbarrow_count = 0
+    for _, item in ipairs(df.global.world.items.other.TOOL) do
+        if df.item_toolst:is_instance(item) then
+            local tool_def = dfhack.items.getSubtypeDef(item:getType(), item:getSubtype())
+            if tool_def and tool_def.id == "ITEM_TOOL_WHEELBARROW" then
+                wheelbarrow_count = wheelbarrow_count + 1 -- Count each wheelbarrow
+                if item.flags.in_job then
+                    -- Skip if in use
+                elseif item.stockpile then
+                    item.stockpile.id = -1 -- Unassign wheelbarrow from stockpile
+                end
+            end
+        end
+    end
+    return wheelbarrow_count -- Return the total count of wheelbarrows processed
+end
+
+local function set_wheelbarrows_for_all_stockpiles()
+    for _, building in ipairs(df.global.world.buildings.all) do
+        if building:getType() == df.building_type.Stockpile then
+            local stockpile_settings = building.settings
+            local skip_wheelbarrows = false
+
+            -- Check food settings to skip wheelbarrows
+            if stockpile_settings and stockpile_settings.food then
+                local food_settings = stockpile_settings.food
+                skip_wheelbarrows = #food_settings.meat > 0 or #food_settings.fish > 0 or #food_settings.unprepared_fish > 0 or
+                                    #food_settings.egg > 0 or #food_settings.plants > 0 or #food_settings.drink_plant > 0 or
+                                    #food_settings.drink_animal > 0 or #food_settings.cheese_animal > 0 or #food_settings.cheese_plant > 0 or
+                                    #food_settings.seeds > 0 or #food_settings.leaves > 0 or #food_settings.powder_plant > 0 or
+                                    #food_settings.powder_creature > 0 or #food_settings.glob > 0 or #food_settings.glob_paste > 0 or
+                                    #food_settings.glob_pressed > 0 or #food_settings.liquid_plant > 0 or #food_settings.liquid_animal > 0 or
+                                    #food_settings.liquid_misc > 0
+                    --dfhack.println("Skipping stockpile due to food items" .. tostring(building))
```

Comment:
I understand what you're saying, but I don't like it. The main purpose of creating this program was to maintain high efficiency in stockpiles containing wheelbarrows and, of course, have wheelbarrows move freely between stockpiles. Having food and stones together sounds insane to me, and this program will instead be a detriment to users, as those stockpiles will not be set to 1 automatically if stone is found or to x (wheelbarrows in the fort) per the program's functionality. Additionally, it will be a problem if 30 wheelbarrows inhabit a food and stone stockpile, as that is a possibility if I remove the conditions, which will cascade into a problem with food. I'm sorry for being stubborn; I think these conditions are essential. If this program is scrapped or repurposed, that's fine.

Reply:
I think you missed the main point of my remark. I wasn't suggesting that this script should fully account for mixed food and stone stockpiles - I agree that those aren't very useful. 

My point was that the program logic could be drastically simplified using a positive check for item categories that require wheelbarrows (i.e. boulders). I don't know which other item categories consistently reach the 75Γ threshold for being carried using a wheelbarrow. Platinum and gold bars do, as do filled minecarts.

Another possibility might be to never check the item flags of the stockpile, and instead only modify the number of requested wheelbarrows for stockpiles that request at least one wheelbarrow. By default, that applies only to stone stockpiles, but the user could decide to assign in wheelbarrows to other stockpiles as well.